pkg/apis/triggers/v1alpha1/trigger_types.go

```diff
@@ -82,25 +83,67 @@ type Trigger struct {
 
 // TriggerInterceptor provides a hook to intercept and pre-process events
 type TriggerInterceptor struct {
-	Webhook   *WebhookInterceptor   `json:"webhook,omitempty"`
-	GitHub    *GitHubInterceptor    `json:"github,omitempty"`
-	GitLab    *GitLabInterceptor    `json:"gitlab,omitempty"`
-	CEL       *CELInterceptor       `json:"cel,omitempty"`
-	Bitbucket *BitbucketInterceptor `json:"bitbucket,omitempty"`
+	// Optional name to identify the current interceptor configuration
+	Name *string `json:"name,omitempty"`
+	// Ref refers to the Interceptor to use
+	Ref InterceptorRef `json:"ref"`
+	// Params are the params to send to the interceptor
+	Params []InterceptorParams `json:"params,omitempty"`
+
+	// Deprecated old fields below
+	Webhook             *WebhookInterceptor   `json:"webhook,omitempty"`
```

Comment:
Should this also be `DeprecatedWebhook`?

Reply:
We need to deprecate Webhooks too but that will be a more involved process since we cannot automatically upgrade existing webhooks to the new API. So, I decided to keep Webhook as non deprecated. (though i put the comment in the wrong spot 😓 )